base/applications/regedit/childwnd.c

```diff
@@ -634,7 +648,15 @@ LRESULT CALLBACK ChildWndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lPa
                         wID = ID_TREE_SUGGESTION_MIN;
                         while(*s && (wID <= ID_TREE_SUGGESTION_MAX))
                         {
-                            _snwprintf(buffer, ARRAY_SIZE(buffer), resource, s);
+                            WCHAR *path = s, buf[MAX_PATH];
+                            if (hRootKey == HKEY_CURRENT_USER || hRootKey == HKEY_LOCAL_MACHINE)
+                            {
+                                // Windows 10 only displays the root name
+                                LPCWSTR next = PathFindNextComponentW(s);
+                                if (next > s)
+                                    lstrcpynW(path = buf, s, min(next - s, _countof(buf)));
```

Comment:
Isn't it possible to avoid the copy into the buffer, and instead, retrieve the necessary beginning pointer and necessary length, and use that in the snwprintf call below?

Reply:
No, that is formatting "Go to %s" in that call (where %s is the suggestion path)

Reply:
Ah yes indeed, my suggestion would have worked only if the formatter was `%.*s`, but that's not the case.